.github/workflows/labelling-bot-deploy.yaml

```diff
@@ -0,0 +1,63 @@
+name: labelling-bot-deploy
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - labelling_bot/**
+
+env:
+  IMAGE_LABELLING_BOT: labelling-bot
+
+jobs:
+  build-and-push-and-deploy-bot:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: labelling_bot/
+          file: labelling_bot/Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKER_HUB_USERNAME }}/${{ env.IMAGE_LABELLING_BOT }}:latest
+      - name: Docker pull and run from Hub
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ secrets.LABELLING_BOT_HOST }}
+          username: ${{ secrets.LABELLING_BOT_SSH_USER }}
+          key: ${{ secrets.LABELLING_BOT_SSH_PRIVATE_KEY }}
+          script: |
+            # Install Docker if not already installed
+            if ! command -v docker &> /dev/null; then
+              sudo yum update -y
+              sudo yum install docker -y
+              sudo service docker start
+              sudo usermod -a -G docker ec2-user
+            fi
+            
+            # Start Docker service if not running
+            if ! sudo service docker status &> /dev/null; then
+              sudo service docker start
+            fi
```

Comment:
why do you need to do this every run? what is the reason to use our own VM for this?

Reply:
Just checking if docker is ready on target instance as it was not ready during first ssh deployment :)